source/CardGames.py

```diff
@@ -235,23 +260,169 @@ def printHands(self, players: dict):
     self.playerinfo = []
-    self.players = players
-    self.value_wanted = int
-    self.suit_wanted = ""
-    self.player_asked = ""
-    self.player_turn = ""
-    self.info_player_turn = int
-    self.info_player_asked = int
+    self.players = []
+    self.player_dict = {}
+    self.value_wanted = 0
+    self.player_asked = None
+    self.player_turn = None
+    self.info_player_turn = 0
+    self.info_player_asked = 0
+    self.deck = Deck()
+    self.dealer = Dealer(self.deck)
+    self.width = os.get_terminal_size().columns
 
   def start_game(self):
-    deck = Deck()
-    dealer = Dealer(deck)
-    dealer.dealCards(5, self.players)
+    
+    print("*._."*(self.width//4))
+    print()
+    print("Welcome to Go Fish!".center(self.width))
+    print()
+    print("*._."*(self.width//4))
+
+    player_number = self.multiplayer_info()
+    self.point_tracker_interface(player_number)
+
+    
+    self.dealer.dealCards(5, self.players)
     for player in self.players:
+      print("\n",player)
       player.showHand(player.hand)
+
+    self.game_loop()
+
+  def game_loop(self):
+      repeat = True
+      while self.count_cards() != True:
+          for key in self.players:
+            turn_repeat = self.ask_card(key)
+            while turn_repeat == "go_again":
+              turn_repeat = self.ask_card(key)
+
+              print()
+
+  def ask_card(self, player_turn):
+    width = os.get_terminal_size().columns
+    deco = "*-"*(width//2)
+
+    self.player_deck_info()
+    
     
+    print(deco)
+    print()
+
+    print(("%s it's your turn!" %player_turn).center(width))
+    player_asked = input("%s who would like to ask? " % str(player_turn))
+
+    print()
+   
+    self.value_wanted = int(input("What value would you like? "))
+    self.info_player_turn = self.playerinfo[self.index(player_turn, self.playerinfo)][1]
+    self.info_player_asked = self.playerinfo[self.index(player_asked, self.playerinfo)][1]
+    self.player_turn = self.playerinfo[self.index(player_turn, self.playerinfo)][0]
+    self.player_asked = self.playerinfo[self.index(player_asked, self.playerinfo)][0]
+    turn_value = self.sayGoFish()
+      
+    self.surrender_card()
+
+    return turn_value
+    
+  def confirm_value(self):
+    confirm_value = True
+    while confirm_value:
+      confirm = input("is this correct? Y/N: ")
+      if confirm == "n":
+        num_players = int(input("How many players? "))
+      elif confirm == "y":
+        confirm_value = False
+      else:
+        print("\n[Please input Y/N]\n")
+          
+  def multiplayer_info(self):
+    width = os.get_terminal_size().columns
+    print()
+    print("*-"*(width//2))
+  
+    num_players = int(input("How many players? "))
+
+    print()
+
+    print("%d players selected" %num_players, "\n")
+
+    self.confirm_value()
+        
+    for i in range(1, num_players+1):
+      print()
+      player = input("Player %d What is your name? " %i)
+      self.players.append(Player(player))
+
```

Comment:
wow lots of functions here! Multiplayer worked very well when tested! great job!

Reply:
Thanks for the feedback!